### DISCUSSION TITLE
Getting an error that says "invalid opcode, The execution might have thrown OR the EVM version used by the selected environment is not compatible with the compiler EVM version."

### DISCUSSION THREAD

**Discussion by Jizzie10** [Cyfrin/foundry-full-course-cu#2153] [General]
I need some assistance with an issue I'm encountering. I'm getting an error in the getter function of a string in my deployed contract that says
"invalid opcode
	The execution might have thrown OR the EVM version used by the selected environment is not compatible with the compiler EVM version.
You may want to cautiously increase the gas limit if the transaction went out of gas." 

I have checked the version for any necessary corrections but it's still giving me errors. Someone should kindly assist.

<img width="1435" alt="Screenshot 2024-08-02 at 00 59 37" src="https://github.com/user-attachments/assets/2b003a4b-3711-4bc1-aed1-38caef671589">

**Comment by EngrPips** [comment]
Hello @Jizzie10, Could you delete the contract and redeploy it? Be sure to know the `EVM` version you used while deploying, then still be on that `EVM` version when you want to interact with the contract. Let's see if the problem persists

**Reply by Jizzie10** [reply]
Oh yes, I was able to rectify the issue. The error occurred with the EVM version: Cancun. After refreshing, it changed to EVM version: Shanghai. I'm a bit confused—could these issues be due to new updates to Remix? I've noticed that the Cancun version doesn't work as well as it did before.

Thanks for the assistance.

**Reply by EngrPips** [reply]
You are welcome, and I am glad you figured it out already. Things change from time to time, and it could be due to the update, but I am not entirely sure.
